src/fsharp/ErrorLogger.fs

```diff
@@ -518,10 +518,7 @@ type OperationResult<'T> =
     
 type ImperativeOperationResult = OperationResult<unit>
 
-let ReportWarnings warns = 
-    match warns with 
-    | [] -> () // shortcut in common case
-    | _ -> List.iter warning warns
+let inline ReportWarnings warns = List.iter warning warns
```

Comment:
the [] check is an optimization that cuts out a couple of stack frames, before the check can occur.  Looking at this, it doesn't seem to occur often enough to warrant either the optimization or the in-line.  Can we lose the inline too?